Mathlib/Topology/UniformSpace/CoveringProfinite.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2025 David Loeffler. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Loeffler
+-/
+import Mathlib.Topology.Algebra.Indicator
+import Mathlib.Topology.Separation.DisjointCover
+import Mathlib.Order.Disjointed
+import Mathlib.Topology.ContinuousMap.Algebra
+
+/-!
+# Uniform approximation by locally constant functions
+
+We show that if `X` is a profinite space, and `V` is a uniform space, then any continuous function
+`X → V` can be uniformly approximated by locally constant functions.
+-/
+
+open UniformSpace Set
+
+open TopologicalSpace (Opens Clopens IsOpenCover)
+
+open scoped Uniformity Function
+
+namespace ContinuousMap
+
+variable {X Y V : Type*}
+  [TopologicalSpace X] [TotallyDisconnectedSpace X] [T2Space X] [CompactSpace X]
+  [UniformSpace V] (f : C(X, V)) {S : Set (V × V)}
+
+/--
+For any continuous function on `X` and `S ∈ 𝓤 V`, there exists a finite cover of `X` by
+pairwise-disjoint nonempty clopens, on each of which `f` varies within `S`.
+-/
+lemma exists_disjoint_nonempty_clopen_cover_of_mem_uniformity (hS : S ∈ 𝓤 V) :
```

Comment:
I understand that this choice is dictated by your previous design in `Separation.DisjointCover`, and the answer is probably there (in which case, I apologise), but why do you prefer `n : Nat` and `Fin n` instead of a more general `I : Type` and `Finite I`? One advantage would be that you could weakend the compactness assumption to extract a cover of `X` indexed by some `I` on which `f` is `S`-constant, and then prove that `Finite I` once you add `CompactSpace X`.

Reply:
I'm uneasy about having statements of the form "there exists a type such that..." because of foundational questions – which type universe should the index type `I` live in?

I am not sure the result is true if one drops the requirement that `X` be compact and allows infinite index sets. If it is true, then it would need a very different proof than the one here.

Reply:
Concerning replacing `Fin n` with `I`, I agree that this makes little sense, in particular since you're constructing one (had it been an assumption, I would probably still argue for some generalisation), so let's forget about it. Concerning compactness, I was simply suggesting to split the proof in two, one the provides a cover where the function is constant (terminating where you've got the `hUC`), and a second where the cover is proven finite under the assumption that `X` be compact.